File: Programmers/java/LEVEL_1_로또의_최고_순위와_최저_순위.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        Set<Integer> set = new HashSet<>();

        for (int n : win_nums) {
            set.add(n);
        }

        int correctCnt = 0;
        int zeroCnt = 0;
        for (int i = 0; i < lottos.length; i++) {
            if (lottos[i] == 0) {
                zeroCnt++;

                continue;
            }

            if (set.contains(lottos[i])) {
                correctCnt++;
            }
        }

        //순위 세팅
        Map<Integer, Integer> map = new HashMap<>();

        for (int i =1; i <= 6; i++) {
            map.put(7 - i, i);
        }
        map.put(0, 6);

        answer[0] = map.get(correctCnt + zeroCnt); //최고 순위
        answer[1] = map.get(correctCnt); //최저 순위
        return answer;
    }
}

File: Programmers/cpp/LEVEL_1_로또의_최고_순위와_최저_순위.cpp
#include <string>
#include <vector>
#include <set>
#include <unordered_map>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
	vector<int> answer;
	set<int> st(win_nums.begin(), win_nums.end());

	int correctCnt = 0;
	int zeroCnt = 0;
	for (int i = 0; i < lottos.size(); i++) {
		if (lottos[i] == 0) {
			zeroCnt++;

			continue;
		}

		if (st.find(lottos[i]) != st.end()) {
			correctCnt++;
		}
	}

	//순위 세팅
	unordered_map<int, int> um; //당첨개수, 순위

	for (int i = 1; i <= 6; i++) {
		um[7 - i] = i;
	}
	um[0] = 6;

	answer.push_back(um[correctCnt + zeroCnt]); //최고 순위
	answer.push_back(um[correctCnt]); //최저 순위
	return answer;
}